lib/spack/spack/spec.py

```diff
@@ -740,9 +744,126 @@ def __str__(self):
             ["^" + self[name].format() for name in sorted(self.keys())])
 
 
+def _libs_default_handler(descriptor, spec, cls):
+    """Default handler when looking for 'libs' attribute. The default
+    tries to search for 'lib{spec.name}' recursively starting from
+    `spec.prefix`.
+
+    :param ForwardQueryToPackage descriptor: descriptor that triggered
+        the call
+    :param Spec spec: spec that is being queried
+    :param type(spec) cls: type of spec, to match the signature of the
+        descriptor `__get__` method
+    """
+    name = 'lib' + spec.name
+    shared = '+shared' in spec
+    return find_libraries(
```

Comment:
just realized that this would work nicely both for `prefix/lib` and `prefix/lib64` and thereby would solve https://github.com/LLNL/spack/issues/2617 :+1: